tools/vendor_hid_test.py

```diff
@@ -265,5 +271,87 @@ def test_11_batch_send_and_interleaved_receive_vendor_first(self):
                                                   self.fido_hid)
 
 
+def get_fido_device() -> CtapHidDevice:
+  for d in CtapHidDevice.list_devices():
+    if d.descriptor.vid == _OPENSK_VID and d.descriptor.pid == _OPENSK_PID:
+      return d
+  raise Exception('Unable to find Fido device')
+
+
+class CliInteraction(UserInteraction):
+  """Sends cancel messages while prompting user."""
+
+  def __init__(self, cid=None, connection: CtapHidConnection = None):
+    super(CliInteraction).__init__()
+    self.cid = cid
+    self.connection = connection
+
+  def prompt_up(self) -> None:
+    print('\n Don\'t touch your authenticator device now...\n')
+    # Send cancel messages to the specified device.
+    if self.connection and self.cid:
+      cancel_packet = (
+          self.cid.to_bytes(4, byteorder='big') + b'\x91' +
+          b''.join([b'\x00'] * 59))
+      assert len(cancel_packet) == _PACKET_SIZE, (
+          f'Expected packet to be {_PACKET_SIZE} '
+          'but was {len(cancel_packet)}')
+
+      self.connection.write_packet(cancel_packet)
+
+
+class CancelTests(unittest.TestCase):
+  """Tests for the canceling while waiting for user touch."""
+
+  @classmethod
+  def setUpClass(cls):
+    cls.fido = get_fido_device()
+
+  def setUp(self) -> None:
+    super().setUp()
+    server = Fido2Server({
+        'id': 'example.com',
+        'name': 'Example RP'
+    },
+                         attestation='direct')
```

Comment:
Indentation here is off? But our Python style checkers don't seem to catch it, curious what's going on here.

Reply:
This is the output of yapf. As it does look weird, I ran autopep8 and it changed the indenting a little.

I'd prefer not to worry too much about the output of auto-formatting..

Reply:
OK, that autopep8 formatting doesn't pass linting. :-(

Going back to original formatting that is done by yapf.